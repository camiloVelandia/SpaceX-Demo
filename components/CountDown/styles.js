import styled from "styled-components";

export const Section = styled.section`
  display: flex;
  justify-content: center;
  align-items: center;
  min-height:700px;
  height:100vh;
  width:100%;
  background-image:url('./images/Dragon.jpg');
  background-position:top;
  background-size:cover;
  background-repeat:no-repeat;
  &::after{
    content:'';
    position:absolute;
    top:0;
    left:0;
    right:0;
    bottom: 0;
    background-color:rgba(0,0,0,.7);
    z-index:0;
  }
`;

export const Counter = styled.div`
  position:relative;
  display: flex;
  flex-direction:column;
  border-radius: 25px;
  background-color: rgba(0, 0, 0, 0.8);
  z-index:1;
  padding:60px 80px;
  & h1{
    font-size:48px;
    text-transform:uppercase;
    text-align:center;
    margin:0;
  }
`;

export const Info = styled.div`
  
  justify-content: space-between;
  padding-top: 40px;
  text-align:center;
  & p{
    margin:0;
    padding-bottom:10px;
    text-transform:capitalize;
  }
  & span{
    text-transform:uppercase;
    color: #00ab66;
  }
`;
export const NumberContainer = styled.div`
  display: flex;
  justify-content:space-between;
  padding-top:40px;
`;

export const Number = styled.div`
  position:relative;
  display: flex;
  flex-direction:column;
  align-items:center;
  justify-content:center;
  z-index:1;
  & p{
    margin:0;
    font-size:80px;
    padding-left:40px;
    padding-right:40px;
  }
  & small{
    margin:0;
    font-size:24px;
    text-transform:uppercase;

  }
`;
