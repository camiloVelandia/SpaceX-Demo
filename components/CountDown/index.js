import React,{useState, useEffect} from 'react';
import { Section, Counter, Number, NumberContainer, Info } from "./styles";
import getNextLaunch from "../../services/getNextLaunch";
import getOneRocket from "../../services/getOneRocket";


const CountDown = () => {
  const [nextlaunch, setNextlaunch] = useState({});
  const [rocketUsed, setRocketUsed] = useState({});
  const [date, setDate] = useState('');
  

  useEffect(() => {
    getNextLaunch()
    .then((data) =>{ 
      setNextlaunch({...data})
      if (nextlaunch.rocket) {
        return getOneRocket(nextlaunch.rocket).then((data) =>
        setRocketUsed({ ...data })
        )
      }
    })
    
    setDate(nextlaunch.date_local);
    
  }, []);

  
  const calculateTimeLeft = (date) => {
      // let year = new Date().getFullYear();
      const difference = +new Date(date) - +new Date();
      let timeLeft = {};
      
      if (difference > 0) {
        timeLeft = {
          days: Math.floor(difference / (1000 * 60 * 60 * 24)),
          hours: Math.floor((difference / (1000 * 60 * 60)) % 24),
          minutes: Math.floor((difference / 1000 / 60) % 60),
          seconds: Math.floor((difference / 1000) % 60),
        };
      }
      
      return timeLeft;
    };
    
    const [timeLeft, setTimeLeft] = useState(calculateTimeLeft());

    useEffect(() => {
      setTimeout(() => {
        setTimeLeft(calculateTimeLeft());
      }, 1000);
    });

    const timerComponents = [];

    Object.keys(timeLeft).forEach((interval) => {
      if (!timeLeft[interval]) {
        return;
      }

      timerComponents.push(
        <span>
          {timeLeft[interval]} {interval}{" "}
        </span>
      );
    });

    console.log(nextlaunch)
  return (
    <Section>
      <Counter>
        <h1>Up Next</h1>

        <NumberContainer>
          <Number>
            <p>3</p>
            <small>days</small>
          </Number>
          <Number>
            <p>
              3
              {timerComponents.length ? (
                timerComponents
              ) : (
                <span>Time's up!</span>
              )}
            </p>
            <small>days</small>
          </Number>
          <Number>
            <p>3</p>
            <small>days</small>
          </Number>
          <Number>
            <p>3</p>
            <small>days</small>
          </Number>
        </NumberContainer>
        <Info>
          <p>
            <span>mission:</span> {nextlaunch.name}
          </p>
          <p>
            <span>Rocket:</span> {rocketUsed.name}
          </p>
        </Info>
      </Counter>
    </Section>
  );
};

export default CountDown;