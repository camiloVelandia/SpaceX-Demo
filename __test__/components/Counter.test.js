import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import CountDown from "../../components/CountDown";

describe("<CountDown/>", () => {
  const countDown = mount(<CountDown />);
  test("render del componente countDown", () => {
    expect(countDown.length).toEqual(1);
  });
});

describe("countDown snaaapshot", () => {
  test("Comprobar ui del countDown", () => {
    const countDown = create(<CountDown />);
    expect(countDown.toJSON()).toMatchSnapshot();
  });
});
