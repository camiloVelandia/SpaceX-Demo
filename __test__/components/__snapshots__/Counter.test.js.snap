// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`countDown snaaapshot Comprobar ui del countDown 1`] = `
<section
  className="styles__Section-k0nii2-0 hPQerN"
>
  <div
    className="styles__Counter-k0nii2-1 jDZgkK"
  >
    <h1>
      Up Next
    </h1>
    <div
      className="styles__NumberContainer-k0nii2-3 ecqJZR"
    >
      <small>
        Time's up!
      </small>
    </div>
    <div
      className="styles__Info-k0nii2-2 jWXqnl"
    >
      <p>
        <span>
          mission:
        </span>
         
      </p>
      <p>
        <span>
          Rocket:
        </span>
         
        falcon
      </p>
    </div>
  </div>
</section>
`;
